.github/workflows/release-main.yml

```diff
@@ -90,3 +90,70 @@ jobs:
           commit-message: "feat: bump kib to ${{ needs.release-please.outputs.release_tag_name }}"
           path: cluster-api-provider-preprovisioned
           title: "feat: bump kib to ${{ needs.release-please.outputs.release_tag_name }}"
+
+  sign-binary:
+    runs-on: macos-latest
+    needs: release-please
+    steps:
+      - if: ${{ needs.release-please.outputs.release_created }}
+        uses: dsaltares/fetch-gh-release-asset@1.0.0
+        with:
+          version: "tags/${{ needs.release-plese.outputs.release_tag_name }}"
+          file: "konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz"
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Create directory to extract the pulled file into
+        run: mkdir "konvoy-image-bundle_darwin_amd64"
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Extract the pulled file
+        run: tar -xvzf "konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz" -C "konvoy-image-bundle_darwin_amd64"
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Remove original downloaded file
+        run: rm "konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz"
+
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Fetch current main directory
+        uses: actions/checkout@v3
+        with:
+          ref: main
+          fetch-depth: 0
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Import Code-Signing Certificates
+        uses: Apple-Actions/import-codesign-certs@v1
+        with:
+          # The certificates in a PKCS12 file encoded as a base64 string
+          p12-file-base64: ${{ secrets.D2IQ_APPLE_DEVELOPER_CERTIFICATE_P12_BASE64 }}
+          # The password used to import the PKCS12 file.
+          p12-password: ${{ secrets.D2IQ_APPLE_DEVELOPER_CERTIFICATE_PASSWORD }}
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Install required binaries
+        run: |
+          brew tap mitchellh/gon
+          brew install mitchellh/gon/gon
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Run Gon to notarize
+        env:
+          GITHUB_TOKEN: ${{ secrets.MESOSPHERECI_USER_TOKEN }}
+          AC_USERNAME: ${{ secrets.D2IQ_APPLE_DEVELOPER_USERNAME }}
+          AC_PASSWORD: ${{ secrets.D2IQ_APPLE_DEVELOPER_PASSWORD }}
+        run: gon ./gon.hcl -log-level=info -log-json ./config.hcl
+
+      - if: ${{ needs.release-please.outputs.release_created }}
+        name: Tar signed bundle
+        run: tar czf konvoy-image-bundle-${{ needs.release-plese.outputs.release_tag_name }}_darwin_amd64.tar.gz konvoy-image-bundle_darwin_amd64
```

Comment:
you will have to regenerate checksum file and upload it. I think @alejandroEsc did wrote that script. cant find it.